modules/spechla/1.0/spechla.smk

```diff
@@ -0,0 +1,310 @@
+        sample_id = "{sample_id}",
+        genome = _which_genome,
+        outdir = CFG["dirs"]["hla_reads"] + "{seq_type}--{genome_build}/"
+    conda: CFG["conda_envs"]["spechla"]
+    group: "extract_and_run"
+    threads:
+        CFG["threads"]["extract_reads"]
+    resources:
+        **CFG["resources"]["extract_reads"]  
+    shell:
+        op.as_one_line("""
+        bash {params.scriptdir}/script/ExtractHLAread.sh -s {params.sample_id} -b {input.bam} -r {params.genome} -o {params.outdir} > {log.stderr} 2>&1
+        """)
+
+#Choose full-length or exon typing [0|1]. 0 indicates full-length, 1 means exon,
+#default is 0. With Exome or RNA data, must select 1 (i.e., exon typing).
+def _which_seqtype(wildcards):
+    if any(seq_type in str({wildcards.seq_type}) for seq_type in ['exome', 'capture', 'target', 'mrna']):
+        return 1
+    else:
+        return 0
+
+rule _spechla_hla_typing:
+    input:
+        fq_1 = CFG["dirs"]["hla_reads"] + "{seq_type}--{genome_build}/{sample_id}_extract_1.fq.gz",
+        fq_2 = CFG["dirs"]["hla_reads"] + "{seq_type}--{genome_build}/{sample_id}_extract_2.fq.gz",fastq_unpaired = CFG["dirs"]["hla_reads"] + "{seq_type}--{genome_build}/{sample_id}_extract.unpaired.fq.gz"
+    output:
+        hla_results = CFG["dirs"]["spechla"] + "{seq_type}--{genome_build}/{sample_id}/hla.result.txt",
+        realigned_bam = temp(CFG["dirs"]["spechla"] + "{seq_type}--{genome_build}/{sample_id}/{sample_id}.realign.sort.bam"),
+        realigned_bam_depth = temp(CFG["dirs"]["spechla"] + "{seq_type}--{genome_build}/{sample_id}/{sample_id}.realign.sort.bam.depth"),
+    log:
+        stderr = CFG["logs"]["spechla"] + "{seq_type}--{genome_build}/{sample_id}/extract_reads.stderr.log"
+    params:
+        scriptdir = CFG["dirs"]["inputs"] + "SpecHLA",
+        sample_id = "{sample_id}",
+        outdir = CFG["dirs"]["spechla"] + "{seq_type}--{genome_build}/",
+        which_seqtype = _which_seqtype,
+        var_quality = CFG["options"]["spechla"]["var_qual"],
+        var_depth = CFG["options"]["spechla"]["var_depth"],
+        min_depth_masking = CFG["options"]["spechla"]["min_depth_masking"],
+        long_indels = CFG["options"]["spechla"]["long_indels"],
+        opts = CFG["options"]["spechla"]["opts"]
+    conda: CFG["conda_envs"]["spechla"]
+    group: "extract_and_run"
+    threads:
+        CFG["threads"]["spechla"]
+    resources:
+        **CFG["resources"]["spechla"]  
+    shell:
+        op.as_one_line("""
+        bash {params.scriptdir}/script/whole/SpecHLA.sh 
+        -n {params.sample_id} 
+        -1 {input.fq_1} 
+        -2 {input.fq_2} 
+        -o {params.outdir} 
+        -u {params.which_seqtype} 
+        -j {threads} 
+        -q {params.var_quality} 
+        -s {params.var_depth} 
+        -k {params.min_depth_masking} 
+        -v {params.long_indels} {params.opts} > {log.stderr} 2>&1
+        """)
+        
+# Remove unwanted outputs
+rule _spechla_cleanup:
```

Comment:
I'm inferring that this is here to ensure the temp files get deleted - do they not get deleted without this rule? Does including the bam files as inputs to the next rule `_spechla_generage_freqlist` rule not do the same thing?

Reply:
I just included it here as a separate rule so I could generate all the intermediate files if necessary. If I use the input of this in the next rule, it should also do the same thing I believe. I'll just keep it as a separate rule just to keep it clean